File: Python/binary_tree.py
class Binary_tree:

	class Node:
		def __init__(self,value,left=None,right=None):
			self.value = value
			self.left = left
			self.right = right

		def __repr__(self):
		  return str(self.value)

	def __init__(self):
		self.root = None

	def add(self,value):
		if self.root == None:
			self.root = Binary_tree.Node(value)
		current = self.root
		while current != None:
			if current.value == value:
				return
			elif value > current.value:
				if current.right != None:
					current = current.right
				else:
					current.right = Binary_tree.Node(value)
					return
			elif value < current.value:
				if current.left != None:
					current = current.left
				else:
					current.left = Binary_tree.Node(value)
					return

	def contains(self,value):
		current = self.root
		while current != None:
			if current.value == value:
				return True
			elif value > current.value:
				current = current.right
			elif value < current.value:
				current = current.left
		return False

File: JavaScript/BinaryTree.js
function BinaryTree() {
    function Node(item, leftNode, rightNode) {
        return {
            value: item,
            left: leftNode,
            right: rightNode
        }
    }
    return {
        root: null,
        add: function(item) {
            if (this.root === null) {
              this.root = Node(item, null, null);
              return;
            }
            let current = this.root;
            while (current != null) {
                if (item === current.value) return;
                if (item > current.value) {
                    if (current.right === null) {
                      current.right = Node(item, null, null)
                      return;
                    } else {
                      current = current.right;
                    }
                } else if (item < current.value) {
                    if (current.left === null) {
                      current.left = Node(item, null, null);
                      return;
                    } else {
                      current = current.left;
                    }
                }
            }
            current = Node(item, null, null);
        },
        contains: function(item) {
            let current = this.root;
            while (current != null) {
                if (item === current.value) return true;
                if (item > current.value) {
                    current = current.right;
                } else if (item < current.value) {
                    current = current.left;
                }
            }
            return false;
        }
    }
}